API/Controllers/ProjectController.cs

```diff
@@ -252,41 +261,68 @@ public async Task<IActionResult> UpdateProject(int projectId, [FromBody] Project
                 return NotFound(problem);
             }
 
-            mapper.Map(projectResource, project);
+            User user = await HttpContext.GetContextUser(userService).ConfigureAwait(false);
+            bool isAllowed = userService.UserHasScope(user.IdentityId, nameof(Defaults.Scopes.ProjectWrite));
+
+            if(!(project.UserId == user.Id || isAllowed))
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to edit the project.",
+                    Detail = "The user is not allowed to edit the project.",
+                    Instance = "2E765D18-8EBC-4117-8F9E-B800E8967038"
```

Comment:
GUID is not unique, generate a fresh one using Visual Studio or an online tool please.